support/globalconfig/ingress.go

```diff
@@ -18,12 +18,17 @@ func IngressConfig() *configv1.Ingress {
 }
 
 func ReconcileIngressConfig(cfg *configv1.Ingress, hcp *hyperv1.HostedControlPlane, globalConfig GlobalConfig) {
-	cfg.Spec.Domain = IngressDomain(hcp)
+	cfg.Spec.Domain = IngressDomain(hcp, globalConfig)
 	if globalConfig.Ingress != nil {
 		cfg.Spec = globalConfig.Ingress.Spec
 	}
 }
 
-func IngressDomain(hcp *hyperv1.HostedControlPlane) string {
-	return fmt.Sprintf("apps.%s", BaseDomain(hcp))
+func IngressDomain(hcp *hyperv1.HostedControlPlane, globalConfig GlobalConfig) string {
+	if globalConfig.Ingress != nil && len(globalConfig.Ingress.Spec.Domain) > 0 {
```

Comment:
use the logic here:
https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/oapi/config.go#L92-L97
AppDomain (if set) should take precedence over Domain

Reply:
done